components/org.wso2.carbon.identity.api.server.workflow.engine/org.wso2.carbon.identity.rest.api.server.workflow.engine.v1/src/main/java/org/wso2/carbon/identity/rest/api/server/workflow/engine/v1/core/WorkflowService.java

```diff
@@ -0,0 +1,631 @@
+
+        WorkflowAssociation association = null;
+
+        if (associationBean != null) {
+
+            association = new WorkflowAssociation();
+
+            association.setId(associationBean.getWorkflowId());
+            association.setWorkflowAssociationName(associationBean.getAssociationName());
+            association.setOperationName(associationBean.getEventName());
+            association.setWorkflowName(associationBean.getWorkflowName());
+            association.setAssociationCondition(associationBean.getCondition());
+            association.setIsEnabled(associationBean.isEnabled());
+        }
+        return association;
+    }
+
+    private Parameter setWorkflowImplParameters(String workflowId, String paramName, String paramValue, String qName, String holder){
+
+        Parameter parameter = new Parameter();
+        parameter.setWorkflowId(workflowId);
+        parameter.setParamName(paramName);
+        parameter.setParamValue(paramValue);
+        parameter.setqName(qName);
+        parameter.setHolder(holder);
+        return parameter;
+    }
+
+    /**
+     * Add new workflow
+     *
+     * Request Body: WorkflowCreation
+     * {
+     *   "workflowName": "User Registration Workflow",
+     *   "workflowDescription": "Workflow to approve new user registrations before account activation",
+     *   "workflowTemplate": {
+     *     "name": "MultiStepApproval",
+     *     "steps": [
+     *       {
+     *         "step": 1,
+     *         "options": [
+     *         {
+     *           "entity": "roles",
+     *           "values": "Internal/admin"
+     *         },
+     *         {
+     *           "entity": "users",
+     *           "values": "John"
+     *         }
+     *
+     *         ]
+     *       },
+     *       {
+     *         "step": 2,
+     *         "options": [
+     *         {
+     *           "entity": "roles",
+     *           "values": "Manager"
+     *         },
+     *         {
+     *           "entity": "users",
+     *           "values": ""
+     *         }
+     *
+     *         ]
+     *       }
+     *       ]
+     *   },
+     *   "workflowEngine": "workflowImplSimple",
+     *   "ApprovalTask": "User Registration Approval",
+     *   "ApprovalTaskDescription": "Approval task to validate and approve new user registrations before account activation"
+     * }
+     *
+     * Response: WorkflowSummary
+     * {
+     *     "id": "748159b8-67fe-4030-9348-bf322d32b267",
+     *     "workflowName": "User Registration Workflow",
+     *     "workflowDescription": "Workflow to approve new user registrations before account activation",
+     *     "workflowTemplate": "MultiStepApprovalTemplate",
+     *     "deployment": "workflowImplSimple"
+     * }
+     *
+     * @param workflow  Workflow details
+     * @param id Workflow ID
+     * @throws WorkflowException
+     */
+    public WorkflowSummary addWorkflow(WorkflowCreation workflow, String id) {
```

Comment:
```suggestion
    public WorkflowSummary addWorkflow(WorkflowCreation workflow) {
```

Reply:
Let's remove the ID from the method parameters.

Reply:
Let's pass the workflow object to the service layer, we can generate ID at the service layer.